### DISCUSSION TITLE
<CR>, <BS> match newline/beginning of line

### DISCUSSION THREAD

**Discussion by hbak** [ggandor/leap.nvim#76] [Ideas]
allow user to match newline with `<CR>` and beginning of line with a `<BS>` or some other non-textual input.

**Comment by ggandor** [comment]
> allow user to match newline with \<CR\>

That is already possible, add `"\r\n"` to `equivalence_classes` (but you need to change some of the special keys then).

> beginning of line with a \<BS\> or some other non-textual input

You mean the first column, or the first non-whitespace character?

**Reply by hbak** [reply]
> You mean the first column, or the first non-whitespace character?

First column.   I realize one could easily get to any first column using the bigram that's there, but this would be a sort of companion to the newline and I think would enable jumping to empty lines.

**Comment by ggandor** [comment]
As for jumping to empty lines, I implemented it in a way that is hopefully intuitive enough - you can simply repeat the newline alias: 346e768bf80487404addd3ccb738feeacdf293b1.

In general, assigning keys to certain special (non-bigram) patterns in the default Leap motions is a slippery slope - we might as well handle word beginnings or all the other EasyMotion targets then.
